spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala

```diff
@@ -73,6 +73,44 @@ abstract class AbstractModule[A <: Activity: ClassTag, B <: Activity: ClassTag,
    */
   var gradInput: A = Activity.allocate[A, T]()
 
+
+  private[bigdl] var inputShapeValue: Activity = null
+
+  private[bigdl] var outputShapeValue: Activity = null
+
+  /**
+   * There's no batch dim in the inputShape which just represent a sample record.
+   */
+  def getInputShape(): Activity = {
```

Comment:
what if a module in two different container/model, which shape should we return?